sound/soc/intel/boards/sof_sdw_common.h

```diff
@@ -25,6 +25,7 @@
 enum {
 	SOF_RT711_JD_SRC_JD1 = 1,
 	SOF_RT711_JD_SRC_JD2 = 2,
+	SOF_RT711_JD_SRC_JD2_100K = 3,
```

Comment:
I see, but that duplicates what is in the Realtek patch and our quirks are now broken if we want to support all jack detection modes

#define SOF_RT711_JDSRC(quirk)		((quirk) & GENMASK(1, 0))

enum rt711_jd_src {
 	RT711_JD_NULL,
 	RT711_JD1,
-	RT711_JD2
+	RT711_JD2,
+	RT711_JD2_100K,
+	RT711_JD2_1P8V_1PORT

@bardliao we probably need to change all the bits definitions for those quirks, and use a single definition instead of redefining them.

Reply:
> I see, but that duplicates what is in the Realtek patch and our quirks are now broken if we want to support all jack detection modes
> 
> ```
> #define SOF_RT711_JDSRC(quirk)		((quirk) & GENMASK(1, 0))
> 
> enum rt711_jd_src {
>  	RT711_JD_NULL,
>  	RT711_JD1,
> -	RT711_JD2
> +	RT711_JD2,
> +	RT711_JD2_100K,
> +	RT711_JD2_1P8V_1PORT
> ```
> 
> @bardliao we probably need to change all the bits definitions for those quirks, and use a single definition instead of redefining them.

Oh, I never thought that a codec can have more than 3 JD source. @plbossart Sorry, I didn't get what is using a single definition. Could you share an example?

Reply:
I meant using the rt711.h values instead of redefining a duplicate

Reply:
> I meant using the rt711.h values instead of redefining a duplicate

Thanks @plbossart See https://github.com/thesofproject/linux/pull/3031